fl4health/clients/moon_client.py

```diff
@@ -108,8 +99,77 @@ def update_before_train(self, current_server_round: int) -> None:
         # Save the parameters of the global model
         self.global_model = self.clone_and_freeze_model(self.model)
 
+        # Optimizing betas for the MK-MMD loss has not been done yet
+        self.betas_optimized = False
+
         return super().update_before_train(current_server_round)
 
+    def update_after_step(self, step: int) -> None:
+        if step > 0 and step % self.beta_update_interval == 0:
+            if self.mkmmd_loss_weights and len(self.old_models_list) > 0 and self.global_model:
+                # Get the feature distribution of the old, local and global models with evaluation mode
+                local_distribution, global_distribution, old_distribution = self.update_buffers(
```

Comment:
Maybe worth mentioning in the comment that we only care about the most recent old model for mkmmd?